deploy-board/deploy_board/webapp/cluster_view.py

```diff
@@ -109,9 +109,13 @@ def post(self, request, name, stage):
 
             return HttpResponse("{}", content_type="application/json")
         except NotAuthorizedException as e:
+            environs_helper.remove_env_capacity(
+                request, name, stage, capacity_type="GROUP", data=cluster_name)
             log.error("Have an NotAuthorizedException error {}".format(e))
             return HttpResponse(e, status=403, content_type="application/json")
         except Exception as e:
+            environs_helper.remove_env_capacity(
+                request, name, stage, capacity_type="GROUP", data=cluster_name)
```

Comment:
I think it would be better if we can do the "roll back" in the `finally` block and control the behavior using a flag such as `remove_env_capacity = True`.
That way if we add new exceptions in the future, it's less likely to make a mistake.